pages/[category]/index.tsx

```diff
@@ -1,51 +1,236 @@
-import { useEffect } from 'react'
+import { useEffect, useState } from 'react'
 import Head from 'next/head'
 import Error from 'next/error'
 import { useRouter } from 'next/router'
-
+import { Fab, Modal, Hidden, Paper, Grid } from '@material-ui/core/'
+import { FilterList, List, Room } from '@material-ui/icons/'
+import withWidth, { WithWidth } from '@material-ui/core/withWidth'
 import {
   useMultipleListRecords,
   useConvertedLocationRecords,
   useLanguage,
 } from '../../hooks/'
-import { siteTitle, categories } from '../../constants'
-import { RecordPane, DisplayMap } from '../../components'
-
-const LandingPage = () => {
+import {
+  RecordPane,
+  DisplayMap,
+  CategoryFilters,
+  MobileFilterModal,
+} from '../../components'
+import {
+  categoryCopy,
+  getMatchingRecords,
+  siteTitle,
+  categories,
+  useStyles,
+} from '../../constants/'
+import { Title, Paragraph } from '../../ui'
+const ITEM_HEIGHT = 48
+const ITEM_PADDING_TOP = 8
+const MenuProps = {
+  PaperProps: {
+    style: {
+      maxHeight: ITEM_HEIGHT * 4.5 + ITEM_PADDING_TOP,
+      width: 250,
+    },
+  },
+}
+function getModalStyle() {
+  return {
+    top: '45%',
+    left: '50%',
+    transform: `translate(-50%, -50%)`,
+  }
+}
+const LandingPage = (props: WithWidth) => {
   const { asPath } = useRouter()
   const { language } = useLanguage()
-
+  const classes = useStyles()
+  const [citySelected, setCitySelected] = useState<string[]>([])
+  const [serviceSelected, setServiceSelected] = useState<string[]>([])
+  const [genderSelected, setGenderSelected] = useState<string[]>([])
+  const [languageSelected, setLanguageSelected] = useState<string[]>([])
+  const [filteredResults, setFilteredResults] = useState<any | null>([])
+  const [open, setOpen] = useState(false)
+  const [currentView, setCurrentView] = useState('list')
+  const [modalStyle] = useState(getModalStyle)
+  const activeCopy = categoryCopy[language]
+  const setFunctions = {
+    setServiceSelected,
+    setGenderSelected,
+    setLanguageSelected,
+    setCitySelected,
+  }
   const validCategory = categories[asPath]
-
-  if (!validCategory) return <Error statusCode={404} />
-
-  const displayCategory: string = validCategory[language].category
-  const routeCategory: string = validCategory.english.category.toLowerCase()
-
+  const routeCategory: string = validCategory?.english.category.toLowerCase()
+  const displayCategory: string = validCategory?.[language].category
+  const displayDescription: string = validCategory?.[language].description
   const { fetchedRecords, setFetchedRecords } =
     useMultipleListRecords(routeCategory)
   const { convertedLocRecords, setLocationRecords } =
     useConvertedLocationRecords()
-
   useEffect((): void => {
-    if (fetchedRecords) setLocationRecords(fetchedRecords)
-  }, [fetchedRecords])
-
+    let keywordQuery = serviceSelected.concat(
+      citySelected,
+      genderSelected,
+      languageSelected,
```

Comment:
do you think we might reuse this filter function? if so, it could probably be abstracted into its own hook